### PR TITLE
ci: Add PR lint to repo

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes

_Jira ticket:_ [INTMDB-934](https://jira.mongodb.org/browse/INTMDB-934)

Since the release process is based on Semantic Versioning and [Conventional Commits](https://www.conventionalcommits.org/), projen will decide on the release version based on the merge PRs titles for that release. See [CONTRIBUTING.md](https://github.com/mongodb/awscdk-resources-mongodbatlas/blob/INTMDB-934/CONTRIBUTING.md#pr-title-format).

This PR adds a GitHub action to check that the PR follows [Conventional Commits](https://www.conventionalcommits.org/).

**Summary of the format to use:**
- `fix description of the ticket`: this correlates with [PATCH](http://semver.org/#summary) in Semantic Versioning
- `feat: description of the ticket`: this correlates with [MINOR](http://semver.org/#summary) in Semantic Versioning
- `BREAKING CHANGE`: a commit that has a footer `BREAKING CHANGE:` or appends a `!` after the type/scope, introduces a breaking API change (correlating with [MAJOR](http://semver.org/#summary) in Semantic Versioning). A BREAKING CHANGE can be part of commits of any type.
Examples:
    - `fix!: description of the ticket` 
    - If the PR has `BREAKING CHANGE:` in its description is a breaking change

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code
- [x] I have tested the CDK constructor in a CFN stack. See [TESTING.md](../TESTING.md)

### CODE REVIEW

**Comment by andreaangiolillo** [comment]
I updated [CONTRIBUTING.md](https://github.com/mongodb/awscdk-resources-mongodbatlas/blob/INTMDB-934/CONTRIBUTING.md#pr-title-format) to exaplain the format to use

**Comment by andreaangiolillo** [comment]
For breaking changes, you just need to add `!` in the title such as `feat!(INTMDB-/CLOUDP-): description of the ticket` or add `BREAKING CHANGE:` in the PR description. Keep in mind that this action will still force you to have the format with `fix` or `feat`

**Review by maastha — DISMISSED** [review]

File: .github/workflows/pull-request-lint.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+            style
+          requireScope: false
+      - uses: marocchino/sticky-pull-request-comment@v2
+        # When the previous steps fails, the workflow would stop. By adding this
+        # condition you can continue the execution with the populated error message.
+        if: always() && (steps.lint_pr_title.outputs.error_message != null)
+        with:
+          header: pr-title-lint-error
+          message: |
+            Hey there and thank you for opening this pull request! 👋🏼
```
Comment (gssbzn) [inline .github/workflows/pull-request-lint.yml:45]: hello there

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -11,14 +11,35 @@ After the above steps are completed and we've agreed on a path forward:
...
+
 ## Autoclose stale issues and PRs
 - After 30 days of no activity (no comments or commits are on an issue or PR) we automatically tag it as “stale” and add a message: "This issue has gone 30 days without any activity and meets the project’s definition of ‘stale’. This will be auto-closed if there is no new activity over the next 30 days. If the issue is still relevant and active, you can simply comment with a “bump” to keep it open, or add the “[Status] Not Stale” label. Thanks for keeping our repository healthy!"
 - After 30 more days of no activity we automatically close the issue / PR.
 
 ## Release and Publishing
 ### Manual Release
 1. `projenrc` is set to do manual release.
-2. The release process is based on *Semantic Versioning* and *Conventional Commits* (https://www.conventionalcommits.org/).
+2. The release process is based on *Semantic Versioning* and [*Conventional Commits*](https://www.conventionalcommits.org/).
```
Comment (gssbzn) [inline CONTRIBUTING.md:42]:
```suggestion
2. The release process is based on [*Semantic Versioning*](https://semver.org/) and [*Conventional Commits*](https://www.conventionalcommits.org/).
```

**Review by gssbzn — APPROVED** [review]

**Review by colm-quinn — APPROVED** [review]
LGTM

**Pull request merged**